pkg/build/controller/build/build_controller.go

```diff
@@ -432,9 +491,20 @@ func (bc *BuildController) createBuildPod(build *buildapi.Build) (*buildUpdate,
 		update.setMessage(buildapi.StatusMessageInvalidOutputRef)
 		return update, err
 	}
+	pushSecret := build.Spec.Output.PushSecret
+	// Only look up a push secret if the user hasn't explicitly provided one.
+	if build.Spec.Output.PushSecret == nil {
```

Comment:
one's as good as another and in this case build.Spec.Output.PushSecret is what we really care about.  pushSecret being a temporary/implementation detail makes me prefer checking the "source of truth".